### PR TITLE
build: Make experimental packages compatible with moduleResolution: node16

### PR DESCRIPTION
**Pull request by tylerbutler**
Updates the packages in experimental/ to build ESM using tsc-multi, create valid ESM types, and adds an exports field and testing using arethetypeswrong.

### CODE REVIEW

**Comment by alexvy86** [comment]
The patch to sorted-btree has the changes shown in [this PR](https://github.com/alexvy86/btree-typescript/pull/1).

File: common/build/build-common/tsconfig.test.json [hunk 1]
```diff
@@ -6,6 +6,5 @@
 		"declaration": false,
 		"declarationMap": false,
 		"skipLibCheck": true,
-		"module": "CommonJS",
```
Comment (tylerbutler) [inline common/build/build-common/tsconfig.test.json:9]: Opportunistic fix. This combination of module/moduleResolution isn't supported.

File: packages/dds/tree/api-report/tree.api.md [hunk 2]
```diff
@@ -4,7 +4,10 @@
 
 ```ts
 
+import { FieldKind as FieldKind_2 } from './schemaTypes';
```
Comment (justus-camp) [inline packages/dds/tree/api-report/tree.api.md:7]: What's with all the `_2`?

Comment (tylerbutler) [reply packages/dds/tree/api-report/tree.api.md:7]: The theory that @alexvy86 and I have is that these show up wherever type inference leads to dynamic imports in the d.ts. I think it's a bug somewhere in the toolchain but not sure where. I think @CraigMacomber looked into this at some point.

File: packages/runtime/id-compressor/package.json [hunk 3]
```diff
@@ -11,6 +11,58 @@
...
+		"./public": {
+			"import": {
+				"types": "./lib/id-compressor-public.d.mts",
+				"default": "./lib/index.mjs"
+			},
+			"require": {
+				"types": "./dist/id-compressor-public.d.ts",
+				"default": "./dist/index.js"
+			}
+		}
```
Comment (alexvy86) [inline packages/runtime/id-compressor/package.json:64]: Isn't the `"."` export supposed to be the public one?

**Review by alexvy86 — APPROVED** [review]
One thing to double check below

File: experimental/dds/sequence-deprecated/package.json [hunk 4]
```diff
@@ -34,7 +47,7 @@
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../.prettierignore",
 		"test": "npm run test:mocha",
 		"test:coverage": "c8 npm test",
-		"test:mocha": "mocha --recursive \"dist/test/**/*.spec.js\" -r node_modules/@fluidframework/mocha-test-setup",
+		"test:mocha": "mocha --recursive \"dist/test/**/*.spec.*js\" -r node_modules/@fluidframework/mocha-test-setup",
```
Comment (DLehenbauer) [inline experimental/dds/sequence-deprecated/package.json:50]: Wasn't sure if the extra '*' before 'js' was intentional normalization or a typo?

(In my local repo, I only see 3 other packages that use this pattern.)

Comment (alexvy86) [reply experimental/dds/sequence-deprecated/package.json:50]: I think intentional, and I'm surprised you see so few, I thought we had already been moving to that in many places. Originally to accommodate for packages that hadn't moved to dual-emit and the ones that now produced `.cjs` files (easier to just have the pattern everywhere than to track them individually). Seems like overall we're going back to a `tsc` build for CJS that outputs `.js` files but I'm in favor of keeping this.

Comment (alexvy86) [reply experimental/dds/sequence-deprecated/package.json:50]: Oh, and originally it was also not specifically for tests, but for code coverage settings and something else in package.json that I'm blanking on.

**Review by DLehenbauer — APPROVED** [review]

File: packages/runtime/id-compressor/package.json (hunk 3 above)
Comment (jason-ha) [reply packages/runtime/id-compressor/package.json:64]: Only once all downstream consumers of non-public are using non-default.

**Pull request closed without merging**